File: processor/src/test/resources/ExpectedSimpleInterfaceMutableWorkerDecoration.java
package com.octo.workerdecorator.processor.test.fixture;

import com.octo.workerdecorator.annotation.java.WorkerDecoration;
import java.lang.Override;
import java.lang.Runnable;
import java.lang.String;
import java.util.Date;
import java.util.concurrent.Executor;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class KotlinInterfaceDecorated implements KotlinInterface, WorkerDecoration<KotlinInterface> {
  private final Executor executor;

  @Nullable
  private KotlinInterface decorated;

  public KotlinInterfaceDecorated(@NotNull Executor executor) {
    this.executor = executor;
  }

  @Override
  public void setDecorated(@Nullable KotlinInterface decorated) {
    this.decorated = decorated;
  }

  @Override
  public void pam() {
    executor.execute(new Runnable() {
      @Override
      public void run() {
        if (decorated != null) {
          decorated.pam();
        }
      }
    });
  }

  @Override
  public void jim(@NotNull String arg0, @Nullable Date arg1) {
    executor.execute(new Runnable() {
      @Override
      public void run() {
        if (decorated != null) {
          decorated.jim(arg0, arg1);
        }
      }
    });
  }

  @Override
  @NotNull
  public KotlinInterface asType() {
    return this;
  }
}

File: processor/src/test/resources/ExpectedSimpleInterfaceMutableWorkerDecoration.kt
package com.octo.workerdecorator.processor.test.fixture

import com.octo.workerdecorator.WorkerDecoration
import java.util.Date
import java.util.concurrent.Executor
import kotlin.String

class KotlinInterfaceDecorated(private val executor: Executor) : KotlinInterface,
        WorkerDecoration<KotlinInterface> {
    private var decorated: KotlinInterface? = null

    override fun pam() {
        executor.execute { decorated?.pam() }
    }

    override fun jim(arg0: String, arg1: Date?) {
        executor.execute { decorated?.jim(arg0, arg1) }
    }

    override fun setDecorated(decorated: KotlinInterface?) {
        this.decorated = decorated
    }

    override fun asType(): KotlinInterface = this
}
